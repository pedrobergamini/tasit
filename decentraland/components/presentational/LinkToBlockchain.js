import React from "react";
import PropTypes from "prop-types";
import { Button, Icon } from "native-base";
import {
  showError,
  openURL,
  buildBlockchainUrlFromActionId,
  getNetworkName,
} from "@helpers";

const supportedNetworks = ["ropsten"];

const _openLinkOf = async actionId => {
  const url = await buildBlockchainUrlFromActionId(actionId);
  try {
    await openURL(url);
  } catch (err) {
    showError(`Unable to open action link`);
  }
};

export default function LinkToBlockchain({ actionId }) {
  // Note: Use 'ropsten' network for tests on development env
  const networkName = getNetworkName();
  const isNetworkSupported = supportedNetworks.includes(networkName);

  if (!isNetworkSupported || !actionId) return null;

  const openLink = () => _openLinkOf(actionId);
  return (
    <Button transparent onPress={openLink}>
      <Icon name="eye" />
    </Button>
  );
}

LinkToBlockchain.propTypes = {
  actionId: PropTypes.string,
};
