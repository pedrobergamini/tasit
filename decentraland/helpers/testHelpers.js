import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () =>
    "0x1234567890123456789012345678901234567890123456789012345678901234",
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
};
