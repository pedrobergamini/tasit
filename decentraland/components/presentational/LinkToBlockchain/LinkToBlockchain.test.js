import React from "react";
import { shallow } from "enzyme";
import LinkToBlockchain from "@presentational/LinkToBlockchain";
import { anAction } from "@helpers/testHelpers";

describe("LinkToBlockchain", () => {
  describe("renders the component", () => {
    it("null action", async () => {
      const actionId = null;
      expect(
        shallow(<LinkToBlockchain actionId={actionId} />)
      ).toMatchSnapshot();
    });

    it("valid action", async () => {
      const actionId = await anAction.getId();
      expect(
        shallow(<LinkToBlockchain actionId={actionId} />)
      ).toMatchSnapshot();
    });
  });
});
