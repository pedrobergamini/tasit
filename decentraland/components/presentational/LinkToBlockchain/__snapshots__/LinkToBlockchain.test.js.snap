// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkToBlockchain renders the component null action 1`] = `""`;

exports[`LinkToBlockchain renders the component valid action 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Styled(Icon)
    name="eye"
    style={
      Object {
        "color": "#07FFC2",
      }
    }
  />
</TouchableOpacity>
`;
